Mathlib/Analysis/Complex/IntegerCompl.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.Analysis.Complex.UpperHalfPlane.Basic
+
+/-!
+# Integer Complement
+
+We define the complement of the integers in the complex plane and give some basic lemmas about it.
+We also show that the upper half plane embeds into the integer complement.
+
+-/
+
+open UpperHalfPlane
+
+/--The complement of the integers in `ℂ`. -/
+abbrev Complex.IntegerComplement := (Set.range ((↑) : ℤ → ℂ))ᶜ
+
+namespace Complex
+
+local notation "ℂ_ℤ " => IntegerComplement
+
+lemma IntegerComplement_eq : ℂ_ℤ = {z : ℂ | ¬ ∃ (n : ℤ), z = ↑n} := by
+  rw [IntegerComplement]
+  aesop
```

Comment:
If you turn the equality around (which you *should* do here and elsewhere) this is just `rfl`:
```suggestion
lemma IntegerComplement_eq : ℂ_ℤ = {z : ℂ | ¬ ∃ (n : ℤ), n = z} := rfl
```